os/net/mac/tsch/tsch-slot-operation.c

```diff
@@ -238,10 +238,20 @@ tsch_release_lock(void)
 
 /* Return channel from ASN and channel offset */
 uint8_t
-tsch_calculate_channel(struct tsch_asn_t *asn, uint8_t channel_offset)
+tsch_calculate_channel(struct tsch_asn_t *asn, uint16_t channel_offset, struct tsch_packet *p)
```

Comment:
could we keep `channel_offset` on 8 bits and keep the check that it's no greater than 15? The code below can still read the packetbuf attribute, and assign it to `channel_offset` only if != 0xffff

Reply:
Channel offset is already 16 bits in the `struct tsch_link` which is passed as a parameter when this function is called (`tsch_calculate_channel(&tsch_current_asn, current_link->channel_offset, current_packet);`) so restricting it to 8 bits in the parameter seems like an oversight. I would like to keep the change to `uint16_t` (or if you're absolutely sure, change the value in `struct tsch_link` and other places to 8 bits)